translator/internal/internal_to_oc.go

```diff
@@ -0,0 +1,436 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	occommon "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	octrace "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/timestamp"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+const sourceFormat = "ot_internal_trace"
+
+var (
+	defaultTime      = time.Time{} // zero
+	defaultTimestamp = timestamp.Timestamp{}
+	defaultProcessID = 0
+)
+
+func internalToOC(td data.TraceData) consumerdata.TraceData {
```

Comment:
We will have to return []consumerdata.TraceData here. The OC format is not capable of representing multiple batches of spans from different resources like internal data.TraceData. So the only approach that guarantees we don't lose data is to return a slice of `consumerdata.TraceData`.

Reply:
Makes sense if `data.TraceData` can have multiple batches of spans that don't relate to each other. I was thinking that `data.TraceData` supposed to have the same set of spans as `consumerdata.TraceData`. 

I changed it to to return slice.